### PR TITLE
Add categorical colors

### PR DESCRIPTION
**Pull request by origami-z**
Adds foundation colors.
Updated foundation page to show colors.

#2523

### CODE REVIEW

**Review by Fercas123 — DISMISSED** [review]

File: docs/components/ColorBlock.css [hunk 1]
```diff
@@ -6,7 +6,7 @@
   border-color: transparent;
   border-style: var(--salt-container-borderStyle);
   border-width: var(--salt-size-border);
-  border-radius: var(--salt-size-unit);
+  border-radius: var(--salt-palette-corner-weak, 0);
```
Comment (origami-z) [inline docs/components/ColorBlock.css:9]: Remove none standard corner, per Darrin's request.

**Review by dplsek — COMMENTED** [review]
Looks good, colors are correct.

**Review by joshwooding — APPROVED** [review]

File: site/docs/foundations/color.mdx [hunk 2]
```diff
@@ -7,4 +7,12 @@ data:
 
 Colors can be used to communicate meaning, suggest a visual hierarchy and help to distinguish our products from others while providing a cohesive accessible experience.
 
+## Foundation
+
+<FoundationColorView group="foundation" />
+
+## Categorical
```
Comment (mark-tate) [inline site/docs/foundations/color.mdx:14]: maybe one for the glossary... foundation color and categorical color

File: site/src/components/css-display/FoundationColorView.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  const [data, setData] = useState<CssVariableData | null>(null);
+
+  useEffect(() => {
+    const fetchJsonData = () => {
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const data = require("./cssFoundations.json") as CssVariableData;
+      const colorKeys = Object.keys(data).filter((x) =>
+        new RegExp(`^--salt-color-\\w+(-\\d+)?$`).test(x)
+      );
+      console.log({ data, colorKeys });
```
Comment (mark-tate) [inline site/src/components/css-display/FoundationColorView.tsx:87]: anonymous debug ?

**Review by mark-tate — DISMISSED** [review]

File: site/src/components/css-display/FoundationColorView.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
```
Comment (joshwooding) [inline site/src/components/css-display/FoundationColorView.tsx:88]:
```suggestion

```

**Review by joshwooding — APPROVED** [review]

**Review by mark-tate — APPROVED** [review]

**Pull request merged**